nan_typedarray_contents.h

```diff
@@ -0,0 +1,77 @@
+/*********************************************************************
+ * NAN - Native Abstractions for Node.js
+ *
+ * Copyright (c) 2015 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
+
+#ifndef NAN_TYPEDARRAY_CONTENTS_H_
+#define NAN_TYPEDARRAY_CONTENTS_H_
+
+template<typename T>
+class TypedArrayContents {
+ public:
+  NAN_INLINE explicit TypedArrayContents(v8::Local<v8::Value> from) :
+      length_(0), data_(NULL) {
+
+#if defined(V8_MAJOR_VERSION) && (V8_MAJOR_VERSION > 4 ||                      \
+  (V8_MAJOR_VERSION == 4 && defined(V8_MINOR_VERSION) && V8_MINOR_VERSION >= 3))
+
+    if (from->IsArrayBufferView()) {
+      v8::Local<v8::ArrayBufferView> array =
+        v8::Local<v8::ArrayBufferView>::Cast(from);
+
+      const size_t    byteLength = array->ByteLength();
+      const ptrdiff_t byteOffset = array->ByteOffset();
+      v8::Local<v8::ArrayBuffer> buffer = array->Buffer();
+
+      char* data = reinterpret_cast<char*>(buffer->GetContents().Data());
+
+      length_ = byteLength / sizeof(T);
+      data_   = reinterpret_cast<T*>(data + byteOffset);
```

Comment:
I think I'd insert an assert that checks the alignment.  You could use `alignof(T)` here but not in general because it's a C++11 construct (io.js v3 and node v4 imply C++11 support but that's not the case with node v0.10 and v0.12.)

Maybe settle for `assert(reinterpret_cast<uintptr_t>(data + byte_offset) % sizeof(T) == 0)`.  There are perhaps still one or two edge cases that will slip through, though.  I'm thinking of things like performing SSE operations on an array of doubles that only has an alignment of 8.